src/app/main/dashboard/memory-usage/memory-usage.component.ts

```diff
@@ -0,0 +1,110 @@
+import {Component, OnInit, ViewChild} from '@angular/core';
+import {MemoryUsageService} from "./shared/memory-usage/memory-usage.service";
+import {ChartComponent} from "angular2-highcharts";
+
+@Component({
+  selector: 'em-memory-usage',
+  templateUrl: './memory-usage.component.html',
+  styleUrls: ['memory-usage.component.scss'],
+  providers: [
+    MemoryUsageService
+  ]
+})
+export class MemoryUsageComponent implements OnInit {
+  @ViewChild(ChartComponent)
+  private chart: ChartComponent;
+
+  private interval = 1;
+  private history = 600;
+
+  current = {
+    free: 0,
+    used: 0,
+    total: 0
+  };
+
+  private chartColors = ['#ffd740', '#69f0ae', '#7c4dff'];
+  private options: HighchartsOptions = {
+    chart: {
+      width: 448,
+      height: 300,
+      type: "area"
+    },
+    credits: {
+      enabled: false
+    },
+    title: {
+      text: null
+    },
+    yAxis: {
+      title: {
+        text: "memory (GB)"
+      }
+    },
+    xAxis: {
+      title: {
+        text: "Time"
+      }
+    },
+    series: [
+      {
+        name: "used",
+        data: [],
+        color: this.chartColors[1]
+      },
+      {
+        name: "total",
+        data: [],
+        color: this.chartColors[2],
+        type: "line"
+      }
+    ]
+  };
+
+  constructor(private memoryUsage: MemoryUsageService) {
+  }
+
+  ngOnInit() {
+  }
+
+  ngAfterViewInit() {
+    this.memoryUsage.getMemoryUsageInInterval(this.interval).subscribe(memory=> {
```

Comment:
Important! The subscription must be removed when destroying the component otherwise, this generates an memory leak.